### DISCUSSION TITLE
ifm3d-ros for Linux

### DISCUSSION THREAD

**Discussion by wjpq** [ifm/ifm3d#397] [Q&A]
I would like to install ifm3d-ros under Linux, my system is ubuntu 20.04. I follow the steps under this website (https://github.com/ifm/ifm3d-ros/blob/master/ifm3d_ros_driver/doc/building.md) to install, but there will be some problems, please help me solve, thank you.
![1](https://github.com/ifm/ifm3d/assets/134564446/d31598b3-bca8-4105-bfb3-1b13d569726b)
![2](https://github.com/ifm/ifm3d/assets/134564446/675e24d4-2951-415e-b72f-d4b3c3e98a94)

**Comment by lola-masson** [comment]
If you are trying to install the latest official ifm3d-ros release, which is 1.0.1, you need to use ifm3d 0.93.0. If you want to use ifm3d 1.2.6, which is the one you currently have installed, according to the screenshots, you can give a try to the ifm3d-ros branch [dev1.2](https://github.com/ifm/ifm3d-ros/tree/dev1.2) which is a work in progress for a new release which will support newer versions of ifm3d.

**Comment by wjpq** [comment]
Hello，When I use ifm3d 0.93.0, I execute "cmake -DCMAKE_INSTALL_PREFIX=/usr .." There will be an error, what is the reason? How solve。
![微信截图_20230526172214](https://github.com/ifm/ifm3d/assets/134564446/7403a603-7a73-4041-bc10-e9508ec6b719)

**Reply by graugans** [reply]
Are you using the git repo or the zip from the download?

**Reply by wjpq** [reply]
I download it from the tags.As shown in the image.
![941](https://github.com/ifm/ifm3d/assets/134564446/d0634cd6-a358-4768-9e09-39865684cf90)

**Reply by lola-masson** [reply]
Unfortunately there is a known bug for the zip installation for this version of ifm3d. If you build from source directly using the cloned code from GitHub, it should work fine. Get the code with `git clone -b v0.93.0 https://github.com/ifm/ifm3d.git --single-branch`.
If you really need to build from source directly from the zip, there is a fix but it is a bit more involved -- it is better if you can directly use the source from git. Let us know.
